add-on/chrome/modules/AboutLoop.jsm

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+/* Basic code borrowed and adapted from PocketAbout stuff in mozilla-central
+ */
+
+const { interfaces: Ci, results: Cr, manager: Cm, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+// See LOG_LEVELS in Console.jsm. Common examples: "All", "Info", "Warn", & "Error".
+const PREF_LOG_LEVEL = "loop.debug.loglevel";
+
+XPCOMUtils.defineLazyGetter(this, "log", () => {
+  let ConsoleAPI = Cu.import("resource://gre/modules/Console.jsm", {}).ConsoleAPI;
+  let consoleOptions = {
+    maxLogLevelPref: PREF_LOG_LEVEL,
+    prefix: "Loop"
+  };
+  return new ConsoleAPI(consoleOptions);
+});
+
+
+function AboutPage(chromeURL, aboutHost, classID, description) {
+  this.chromeURL = chromeURL;
+  this.aboutHost = aboutHost;
+  this.classID = Components.ID(classID);
+  this.description = description;
+}
+
+AboutPage.prototype = {
+  QueryInterface: XPCOMUtils.generateQI([Ci.nsIAboutModule]),
+  getURIFlags: function(aURI) { // eslint-disable-line no-unused-vars
+    return Ci.nsIAboutModule.ALLOW_SCRIPT |
+           Ci.nsIAboutModule.HIDE_FROM_ABOUTABOUT |
+           Ci.nsIAboutModule.MAKE_UNLINKABLE |
+           Ci.nsIAboutModule.URI_SAFE_FOR_UNTRUSTED_CONTENT |
+           Ci.nsIAboutModule.URI_MUST_LOAD_IN_CHILD;
+           // XXX note that above line applies to panel and convo! ok?
+  },
+
+  newChannel: function(aURI, aLoadInfo) {
```

Comment:
I believe something about this function isn't working. I tried a build of FF where I removed the about urls from its core, and got this when trying to open the conversation window:

```
[Exception... "newChannel requires at least one of the 'loadingNode', 'loadingPrincipal', or 'loadUsingSystemPrincipal' properties on the options object."  nsresult: "0x80070057 (NS_ERROR_ILLEGAL_VALUE)"  location: "JS frame :: resource://gre/modules/RemoteAddonsParent.jsm :: AboutProtocolParent.openChannel :: line 268"  data: no]
(unknown)
Empty string passed to getElementById().remote-browser.xml:232:7
TypeError: can't convert undefined to objectRemoteAddonsChild.jsm:258:9
```
If you try just loading `about:panel` in a tab, then you get a similar issue (whereas it loaded before).

I think we need to do something like:

http://hg.mozilla.org/comm-central/file/tip/mail/components/aboutRedirector.js#l56

I haven't quite figured out why the pocket example doesn't have that, nor why `AboutRedirector.cpp` doesn't need to do something similar. So I'm a bit puzzled there.